addons/im_livechat/models/res_partner.py

```diff
@@ -16,43 +17,48 @@ class ResPartner(models.Model):
         if channel.channel_type != "livechat" or not self:
             return
         lang_name_by_code = dict(self.env["res.lang"].get_installed())
-        invite_by_self_count_by_partner = dict(
+        invite_by_self_count_by_partner = defaultdict(
+            int,
             self.env["discuss.channel.member"]._read_group(
                 [["create_uid", "=", self.env.user.id], ["partner_id", "in", self.ids]],
                 groupby=["partner_id"],
                 aggregates=["__count"],
-            )
+            ),
         )
         active_livechat_partners = (
             self.env["im_livechat.channel"].search([]).available_operator_ids.partner_id
         )
-        for partner in self:
-            languages = list(OrderedSet([
-                lang_name_by_code[partner.lang],
-                # sudo: res.users.settings - operator can access other operators languages
-                *partner.user_ids.sudo().livechat_lang_ids.mapped("name")
-            ]))
-            store.add(
-                partner,
-                {
-                    "invite_by_self_count": invite_by_self_count_by_partner.get(partner, 0),
-                    "is_available": partner in active_livechat_partners,
-                    "lang_name": languages[0],
-                    # sudo: res.users.settings - operator can access other operators expertises
-                    "livechat_expertise": partner.user_ids.sudo().livechat_expertise_ids.mapped("name"),
-                    "livechat_languages": languages[1:],
-                    # sudo: res.users.settings - operator can access other operators livechat usernames
-                    "user_livechat_username": partner.sudo().user_livechat_username,
-                },
-                # sudo - res.partner: checking if operator is in call for live
-                # chat invitation is acceptable.
-                extra_fields=[Store.Attr("is_in_call", sudo=True)]
+        languages_by_partner = {
+            partner: list(
+                OrderedSet(
+                    [
+                        lang_name_by_code[partner.lang],
+                        # sudo: res.users.settings - operator can access other operators languages
+                        *partner.user_ids.sudo().livechat_lang_ids.mapped("name"),
+                    ],
+                ),
             )
+            for partner in self
+        }
+        res.attr("invite_by_self_count", invite_by_self_count_by_partner.get)
+        res.attr("is_available", lambda p: p in active_livechat_partners)
+        res.attr("lang_name", lambda p: languages_by_partner[p][0])
+        # sudo: res.users.settings - operator can access other operators expertises
+        res.attr(
+            "livechat_expertise",
+            lambda p: p.user_ids.sudo().livechat_expertise_ids.mapped("name"),
+        )
+        res.attr("livechat_languages", lambda p: languages_by_partner[p][1:])
```

Comment:
Following the previous comment:

```suggestion
        res.attr("livechat_languages", lambda p: p.user_ids.sudo().livechat_lang_ids.mapped("name"))
```